Source/Hydro.H

```diff
@@ -73,42 +479,504 @@ pc_divu(
+  const amrex::Real surrounding_vol =
+    vfrac(i, j, k) + vfrac(i - 1, j, k) + vfrac(i, j - 1, k) +
+    vfrac(i - 1, j - 1, k) + vfrac(i, j, k - 1) + vfrac(i - 1, j, k - 1) +
+    vfrac(i, j - 1, k - 1) + vfrac(i - 1, j - 1, k - 1);
+  const amrex::Real cutoff = 8.0 * (1.0 - 1.e-12);
+#endif
+
+  divu(i, j, k) = 0.0;
+
+  if (surrounding_vol >= cutoff) {
+
+#if (AMREX_SPACEDIM == 2)
+    const amrex::Real ux = 0.5 *
+                           (q(i, j, k, QU) - q(i - 1, j, k, QU) +
+                            q(i, j - 1, k, QU) - q(i - 1, j - 1, k, QU)) *
+                           dxInv[0];
+    const amrex::Real vy = 0.5 *
+                           (q(i, j, k, QV) - q(i, j - 1, k, QV) +
+                            q(i - 1, j, k, QV) - q(i - 1, j - 1, k, QV)) *
+                           dxInv[1];
+#elif (AMREX_SPACEDIM == 3)
+    const amrex::Real ux =
+      0.25 *
+      (q(i, j, k, QU) - q(i - 1, j, k, QU) + q(i, j, k - 1, QU) -
+       q(i - 1, j, k - 1, QU) + q(i, j - 1, k, QU) - q(i - 1, j - 1, k, QU) +
+       q(i, j - 1, k - 1, QU) - q(i - 1, j - 1, k - 1, QU)) *
+      dxInv[0];
+
+    const amrex::Real vy =
+      0.25 *
+      (q(i, j, k, QV) - q(i, j - 1, k, QV) + q(i, j, k - 1, QV) -
+       q(i, j - 1, k - 1, QV) + q(i - 1, j, k, QV) - q(i - 1, j - 1, k, QV) +
+       q(i - 1, j, k - 1, QV) - q(i - 1, j - 1, k - 1, QV)) *
+      dxInv[1];
+
+    const amrex::Real wz =
+      0.25 *
+      (q(i, j, k, QW) - q(i, j, k - 1, QW) + q(i - 1, j, k, QW) -
+       q(i - 1, j, k - 1, QW) + q(i, j - 1, k, QW) - q(i, j - 1, k - 1, QW) +
+       q(i - 1, j - 1, k, QW) - q(i - 1, j - 1, k - 1, QW)) *
+      dxInv[2];
+#endif
+    divu(i, j, k) = AMREX_D_TERM(ux, +vy, +wz);
+  }
+}
+
+AMREX_GPU_DEVICE AMREX_FORCE_INLINE void
+pc_eb_pdivu(
+  int i,
+  int j,
+  int k,
+  amrex::Array4<amrex::Real const> const& q,
+  AMREX_D_DECL(
+    amrex::Array4<amrex::Real const> const& q1,
+    amrex::Array4<amrex::Real const> const& q2,
+    amrex::Array4<amrex::Real const> const& q3),
+  amrex::Array4<amrex::Real> const& divu,
+  amrex::Array4<amrex::EBCellFlag const> const& flag,
+  amrex::Array4<amrex::Real const> const& vfrc,
+  AMREX_D_DECL(
+    amrex::Array4<amrex::Real const> const& apx,
+    amrex::Array4<amrex::Real const> const& apy,
+    amrex::Array4<amrex::Real const> const& apz),
+  AMREX_D_DECL(
+    amrex::Array4<amrex::Real const> const& fcx,
+    amrex::Array4<amrex::Real const> const& fcy,
+    amrex::Array4<amrex::Real const> const& fcz),
+  amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> const& dxinv)
+{
+  amrex::Real pdivu;
+
+#if (AMREX_SPACEDIM == 2)
+  if (flag(i, j, k).isRegular()) {
+    pdivu = 0.5 * ((q1(i + 1, j, k, GDPRES) + q1(i, j, k, GDPRES)) *
+                     (q1(i + 1, j, k, GDU) - q1(i, j, k, GDU)) * dxinv[0] +
+                   (q2(i, j + 1, k, GDPRES) + q2(i, j, k, GDPRES)) *
+                     (q2(i, j + 1, k, GDV) - q2(i, j, k, GDV)) * dxinv[1]);
+    divu(i, j, k, UEINT) += pdivu;
```

Comment:
@baperry2 I think this (and the others in this function) should be `-= pdivu;` since that would follow what we do for the non-eb path. But flipping the sign doesn't change anything and I even multiplied by 1e12 and nothing happens. Maybe we just don't use this...

Reply:
Going on very old memories, but might the "artificial viscosity" be zero in the cases you're testing with?

Reply:
hmmm this comment sent me into a tailspin. I've got bug fixes out to CAMR as well.  I think the present is going to be late... but it's going to happen!

Reply:
I hear some people give gifts for New Year's.

Reply:
Looks like you resolved this in CAMR, so I'm fine with this if it macthes that solution.